src/write_gcm.c

```diff
@@ -2961,11 +2965,14 @@ static void wg_json_Points(json_ctx_t *jc, const wg_payload_t *element) {
 static int wg_json_CreateTimeSeries(
     json_ctx_t *jc, const monitored_resource_t *resource,
     const wg_payload_t *head, const wg_payload_t **new_head) {
+  // Count tracks the number of payloads. This is the same as the number of time
+  // series because we enforce one value per payload below.
   int count = 0;
 
   wg_json_array_open(jc);
 
-  for (; head != NULL && jc->error == 0; head = head->next) {
+  for (; head != NULL && jc->error == 0 && count < MAX_TIME_SERIES;
```

Comment:
Can you please make this an explicit conditional with a `break` in it, with a comment describing what it does (similar to the `buffer_length` check below)?

Reply:
Done.